modular/approver/approve_task.go

```diff
@@ -98,7 +100,33 @@ func (a *ApprovalModular) PostCreateBucketApproval(ctx context.Context, task cor
 }
 
 func (a *ApprovalModular) PreMigrateBucketApproval(ctx context.Context, task coretask.ApprovalMigrateBucketTask) error {
-	// TODO: check sp id.
+	var err error
+	if task == nil || task.GetMigrateBucketInfo() == nil {
+		log.CtxError(ctx, "failed to ask migrate bucket approval due to bucket info pointer dangling")
+		return ErrDanglingPointer
+	}
+	defer func() {
+		if err != nil {
+			task.SetError(err)
+		}
+		log.CtxDebugw(ctx, task.Info())
+	}()
+	selfSPID, err := a.getSPID()
+	if err != nil {
+		return err
+	}
+	if selfSPID != task.GetMigrateBucketInfo().GetDstPrimarySpId() {
+		return fmt.Errorf("not the correct sp to ask approval")
+	}
+
+	migratingBucketCount, err := a.baseApp.GfSpClient().GetSPMigratingBucketNumber(ctx, selfSPID)
```

Comment:
Does a bucket migration task can be expired?  If an expired task is count here as well, it may cause always approval rejection problem.